rdt/transformers/base.py

```diff
@@ -31,25 +175,59 @@ def transform(self, data):
         """
         raise NotImplementedError()
 
-    def fit_transform(self, data):
-        """Fit the transformer to the data and then transform it.
+    def fit_transform(self, data, columns=None):
+        """Fit the transformer to the `columns` of the `data` and then transform them.
 
         Args:
-            data (pandas.Series or numpy.array):
-                Data to transform.
+            data (pandas.DataFrame):
+                The entire table.
+            columns (list or tuple or str):
+                List or tuple of column names from the data to transform.
+                If only one column is provided, it can be passed as a string instead.
+                If none are passed, fits on the entire dataset.
 
         Returns:
-            numpy.array:
-                Transformed data.
+            pd.DataFrame:
+                The entire table, containing the transformed data.
         """
-        self.fit(data)
+        self.fit(data, columns)
         return self.transform(data)
 
     def reverse_transform(self, data):
         """Revert the transformations to the original values.
 
         Args:
-            data (pandas.Series or numpy.array):
+            data (pandas.DataFrame):
+                The entire table.
+
+        Returns:
+            pandas.DataFrame:
+                The entire table, containing the reverted data.
+        """
+        if isinstance(data, pd.Series):
+            data = pd.DataFrame(data, columns=['series'])
+
+        if len(self._columns) == 1:
+            columns = self._columns[0]
+
+        if len(self._output_columns) == 1:
+            output_columns = self._output_columns[0]
+
+        # what about columns being partially in the data?
+        if any(column not in data.columns for column in self._columns):
+            return data
+
+        columns_data = data[output_columns]
+        data[columns] = self._reverse_transform(columns_data)
+        data.drop(list(self._columns_to_drop), axis=1)
```

Comment:
Here we should be dropping `self._output_columns`: `data.drop(self._output_columns, axis=1)`